og.module

```diff
@@ -117,6 +105,270 @@ function og_entity_insert(EntityInterface $entity) {
+      $access = og_user_access_entity('update group', 'node', $node, $account);
+    }
+
+    if ($access) {
+      return AccessResult::allowed();
+    }
+
+    // Otherwise, ignore or deny based on whether strict entity access is set.
+    return AccessResult::forbiddenIf($entity_access_strict);
+  }
+
+  return AccessResult::neutral();
+}
+
+/**
+ * Determine whether a user has a given privilege.
+ *
+ * All permission checks in OG should go through this function. This
+ * way, we guarantee consistent behavior, and ensure that the superuser
+ * and group administrators can perform all actions.
+ *
+ * @param $group_type
+ *   The entity type of the group.
+ * @param $gid
+ *   The entity ID of the group.
+ * @param string $operation
+ *   The entity operation being checked for.
+ * @param $account
+ *   (optional) The account to check. Defaults to the current user.
+ * @param $skip_alter
+ *   (optional) If TRUE then user access will not be sent to other modules
+ *   using drupal_alter(). This can be used by modules implementing
+ *   hook_og_user_access_alter() that still want to use og_user_access(), but
+ *   without causing a recursion. Defaults to FALSE.
+ * @param $ignore_admin
+ *   (optional) When TRUE the specific permission is checked, ignoring the
+ *   "administer group" permission if the user has it. When FALSE, a user
+ *   with "administer group" will be granted all permissions.
+ *   Defaults to FALSE.
+ *
+ * @return
+ *   TRUE or FALSE if the current user has the requested permission.
+ *   NULL, if the given group isn't a valid group.
+ */
+function og_user_access($group_type, $gid, $operation, $account = NULL, $skip_alter = FALSE, $ignore_admin = FALSE) {
+  $perm = &drupal_static(__FUNCTION__, []);
+  // Mark the group ID and permissions that invoked an alter.
+  $perm_alter = &drupal_static(__FUNCTION__ . '_alter', []);
+
+  if (!Og::isGroup($group_type, $gid)) {
+    // Not a group.
+    return NULL;
+  }
+
+
+  if (empty($account)) {
+    $account = clone \Drupal::currentUser()->getAccount();
+  }
+
+  $account_id = $account->id();
+
+  // User #1 has all privileges.
+  if ($account_id == 1) {
+    return TRUE;
+  }
+
+  // Administer group permission.
+  if ($account->hasPermission('administer group') && !$ignore_admin) {
+    return TRUE;
+  }
+
+  // Group manager has all privileges (if variable is TRUE).
+  if (!empty($account_id) && \Drupal::config('og.settings')->get('group_manager_full_access')) {
+    $group = entity_load($group_type, $gid);
+
+    if (($group instanceof EntityOwnerInterface) && ($group->getOwnerId() == $account_id)) {
+      return TRUE;
+    }
+  }
+
+  $identifier = $group_type . ':' . $gid;
+
+  // To reduce the number of SQL queries, we cache the user's permissions
+  // in a static variable.
+  if (!isset($perm[$identifier][$account_id])) {
+    $perms = array();
+
+    if ($roles = og_get_user_roles($group_type, $gid, $account_id)) {
```

Comment:
tl;dr - Yes. 

long - I think if we have api function for get the roles, for example, they need to be as a static method on the user role entity. Look me to more encapsulate.